File: OpenCredentialPublisher.ObcLibrary/src/OB20/CryptographicKeyDType.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace OpenCredentialPublisher.ObcLibrary.OB20
{
    public class CryptographicKeyDType
    {
        /// <summary>
        /// Unique IRI for this object. Model Primitive Datatype = NormalizedString.
        /// </summary>
        [Required]
        [JsonPropertyName("id")]
        public string Id { get; set; }

        /// <summary>
        /// The JSON-LD type of this object. Normally 'CryptographicKey'. Model Primitive Datatype = String.
        /// </summary>
        [JsonPropertyName("type")]
        public string Type { get; set; }

        /// <summary>
        /// The identifier for the Profile that owns this key. Model Primitive Datatype = String.
        /// </summary>
        [JsonPropertyName("owner")]
        public string Owner { get; set; }

        /// <summary>
        /// The PEM key encoding is a widely-used method to express public keys, compatible with almost every Secure Sockets Layer library implementation. Model Primitive Datatype = String.
        /// </summary>
        [JsonPropertyName("publicKeyPem")]
        public string PublicKeyPem { get; set; }
    }
}

File: OpenCredentialPublisher.Wallet/src/ClientApp/src/app/shared/models/clrLibrary/cryptographicKeyDType.ts
/** Based on the Key class from the W3C Web Payments Community Group Security Vocabulary. A CryptographicKey document identifies and describes a public key used to verify signed Assertions. */
export class CryptographicKeyDType {
  /** URL to the JSON-LD context file. */
  context: string;
  /** The URI of the CryptographicKey document. Used during signed verification. Model Primitive Datatype = NormalizedString. */
  id: string;
  /** The JSON-LD type of this object. Normally 'CryptographicKey'. Model Primitive Datatype = NormalizedString. */
  type: string;
  /** The identifier for the Profile that owns this PUBLIC KEY and the PRIVATE KEY used to sign the assertion or endorsement. Model Primitive Datatype = NormalizedString. */
  owner: string;
  /** The PUBLIC KEY in PEM format corresponding to the PRIVATE KEY used by the owner to sign the assertion or endorsement. The PEM key encoding is a widely-used method to express public keys, compatible with almost every Secure Sockets Layer library implementation. Model Primitive Datatype = String. */
  publicKeyPem: string;
  additionalProperties: { [index: string]: any };
}
